code/_onclick/ai.dm

```diff
@@ -218,3 +225,51 @@
 		message += " It is underground."
 	to_chat(user, span_notice(message))
 
+/turf/AICtrlShiftClick(mob/living/silicon/ai/user)
+	///var to keep us from queuing a million railgun shots and breaking the system, do_afters don't work well with AI actions
+	var/static/busy = null
+	var/laz_name = user.name
+	var/obj/effect/overlay/temp/laser_target/laser
+	var/area/A = get_area(user.eyeobj.loc)
+	if(busy)
+		to_chat(user, span_warning("The rail guns are already targeting a location, wait for them to finish."))
+		return
+	if(locate(/mob/living/carbon/human) in range(AI_RAILGUN_HUMAN_EXCLUSION_RANGE, get_turf(user.eyeobj.loc)))
+		to_chat(user, span_warning("Friendly biologicals detected near [loc] target location, aborting."))
+		return
+	if(!is_ground_level(user.eyeobj.z) || isdropshiparea(A)) //can't fire the railgun off the ground level, or at the DS
+		to_chat(user, span_warning("Incompatible target location."))
+		return
+	if(A.ceiling > CEILING_METAL)
+		to_chat(user, span_warning("DEPTH WARNING: Target too deep for ordnance."))
```

Comment:
some of these tochats could be balloons

Reply:
I tried a bunch of things, including balloon_alert_to_viewers and couldn't do this one. The alert won't follow the eyeobj, even if you directly specify that as where the alert should spawn. I had brief luck with using balloon_alert_to_viewers, but there's a cutoff a certain distance away from the AI core.